### PR TITLE
feat: Learning assistant plugin to add in pluginslots

### PR DESCRIPTION
**Pull request by mraman-2U**
This pull request introduces several improvements and refactors to the Xpert Learning Assistant plugin, focusing on plugin registration, configuration, API response handling, and test coverage. The main themes are plugin system integration, API and state management enhancements, and improved test reliability and coverage.

**Plugin system integration and configuration:**

- Added a new `XpertPlugin` object and registration utility in `src/plugin.js` to formally register the Xpert Learning Assistant as a plugin, including logic for when the assistant should be enabled and how its props are generated. This enables easier integration with the host application's plugin system and improves maintainability.
- Updated the public API in `src/index.jsx` to export the new plugin registration utilities and the `XpertChatWidget` component, supporting both legacy and new plugin registration approaches.
- Introduced a new `src/constants.js` file for enrollment mode constants, reducing cross-app dependencies and centralizing configuration.

**API and state management improvements:**

- Refactored the chat API response handling in `src/data/thunks.js` to use feature flags from `getConfig()` instead of environment variables, improved array/object response detection, and ensured only valid messages are dispatched. Added error logging and improved audit trial validation. [[1]](diffhunk://#diff-f708120055e193e6e1c566fefb57a208b514e2fd0f8944fc91ee1f0c97b0cd34R25-R70) [[2]](diffhunk://#diff-f708120055e193e6e1c566fefb57a208b514e2fd0f8944fc91ee1f0c97b0cd34L73-R132) [[3]](diffhunk://#diff-f708120055e193e6e1c566fefb57a208b514e2fd0f8944fc91ee1f0c97b0cd34L118-L168)

**Testing enhancements:**

- Improved test coverage in `src/data/thunks.test.js` to cover new response handling logic, edge cases, and feature flag scenarios. Tests now use `getConfig()` mocks, and timer mocks are handled more robustly. [[1]](diffhunk://#diff-64688506fa5a6fcbe2cd60fea5be1774aae123fd6fb19fcda6906aa8ce2c7635R4) [[2]](diffhunk://#diff-64688506fa5a6fcbe2cd60fea5be1774aae123fd6fb19fcda6906aa8ce2c7635R19-R23) [[3]](diffhunk://#diff-64688506fa5a6fcbe2cd60fea5be1774aae123fd6fb19fcda6906aa8ce2c7635L28-R60) [[4]](diffhunk://#diff-64688506fa5a6fcbe2cd60fea5be1774aae123fd6fb19fcda6906aa8ce2c7635L66-R75) [[5]](diffhunk://#diff-64688506fa5a6fcbe2cd60fea5be1774aae123fd6fb19fcda6906aa8ce2c7635L78-R191)

**Build and development tooling:**

- Added `@openedx/frontend-build` and `redux-mock-store` to `devDependencies` in `package.json` to support improved build processes and more robust Redux testing.

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/**
+ * Xpert Learning Assistant Plugin Registration
+ *
+ * This file registers the Xpert learning assistant as a plugin
```
Comment (michaelroytman) [inline src/plugin.js]: This isn't how we register a plugin with an MFE plugin slot. You can review [Host MFE JS-based Configuration](https://github.com/openedx/frontend-plugin-framework?tab=readme-ov-file#host-mfe-js-based-configuration) for example. We'd export the top level plugin component and import it from our Javascript configuration and add it to the `pluginSlots` key in the `config` object.

File: src/widgets/XpertChatWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  // Note: This assumes the host MFE exposes the model store in Redux
+  const course = useSelector(state => state.models?.coursewareMeta?.[courseId]);
+  // Get audit trial info from learning assistant slice (populated after summary fetch)
+  const auditTrial = useSelector(state => state.learningAssistant?.auditTrial);
+
+  // Get learning assistant enabled flag from course metadata
+  const learningAssistantEnabled = course?.learningAssistantEnabled;
+
+  // Get content tools setting from config
+  const contentToolsEnabled = config.CONTENT_TOOLS_ENABLED;
```
Comment (michaelroytman) [inline src/widgets/XpertChatWidget.jsx]: Where does this value come from? Whether content tools are enabled is a dynamic value, computed from the course configuration. It cannot be made a static value.

File: src/widgets/XpertChatWidget.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+  // === Debug Logging (Development Only) ===
+
+  // === Conditional Rendering Logic ===
+
+  // 1. Check if learning assistant is enabled for this course
+  // Note: If learningAssistantEnabled is undefined, assume it's enabled
```
Comment (michaelroytman) [inline src/widgets/XpertChatWidget.jsx]: Could you explain your rationale for this? I would think it should not be enabled in this case.

File: src/widgets/XpertChatWidget.jsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
+import React from 'react';
```
Comment (michaelroytman) [inline src/widgets/XpertChatWidget.jsx]: I'm curious why you created a separate component from `Xpert`? Is the purpose of this component to wrap all visibility rules into a single component instead of putting it in `Xpert`?

File: src/constants.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (michaelroytman) [inline src/constants.js]: I feel like it would be better to import these from the learning MFE, since it's the source of truth. What do you think?

**Review by michaelroytman — CHANGES_REQUESTED** [review]
I left some comments and questions.

**Review by michaelroytman — APPROVED** [review]
I see some Copilot comments, but I don't have anything to add on top of those comments.
